examples/cancelable_action/cancelable_action/server/cancelable_action.hpp

```diff
@@ -67,19 +70,18 @@ namespace examples { namespace server
         // Cancel the lengthy action above
         void cancel_it()
         {
-            std::lock_guard<mutex_type> l(mtx_);
-            if (id_ != hpx::thread::id()) {
-                hpx::thread::interrupt(id_);
```

Comment:
hmm, wouldn't it make more sense to unlock `mtx_` before calling `hpx::thread::interrupt(id_)`?
Something like this:
```

hpx::thread_id::id id;
{
    std::lock_guard<mutex_type> l(mtx_);
    std::swap(id, id_);
}
if (id != hpx::thread::id())
{
    hpx::thread::interrupt(id);
}
````

Reply:
Yeah, that could be changed, but it's actually taking the lock in the `reset_id` destructor that causes the problem. To be honest, the example wouldn't even need an atomic (in this particular example and use case).

Reply:
Ok, I haven't looked too closely. Just noticed the slight change of semantics with your proposed changes...